azkaban-common/src/main/java/azkaban/project/ProjectLoader.java

```diff
@@ -200,4 +200,17 @@ Props fetchProjectProperty(int projectId, int projectVer, String propsName)
       throws ProjectManagerException;
 
   void updateProjectSettings(Project project) throws ProjectManagerException;
+
+  /**
+   * Uploads flow file.
+   */
+  void uploadFlowFile(int projectId, int projectVersion, int flowVersion, File flowFile)
```

Comment:
same here about the argument order.

Reply:
I'm ok with either order. I chose this argument order based on the sequence in the project_flow_fiels DB table. 
project_id | project_version | flow_name | flow_version | modified_time | flow_file